status-whisper-usage-spec.md

```diff
@@ -59,6 +60,28 @@ Whisper's Proof Of Work algorithm is used to deter denial of service and various
 * proof-of-work requirement not larger than `0.002`
 * time-to-live not lower than `10` (in seconds)
 
+## Rate limiting
+
+In order to provide an optional very basic Denial-of-Service attack protection, each node SHOULD define its own rate limits. The rate limits SHOULD be applied on IPs, peer IDs, and envelope topics.
+
+Each node MAY decide to whitelist, i.e. do not rate limit, selected IPs or peer IDs.
+
+If a peer exceeds node's rate limits, the connection between them MAY be dropped.
+
+Each node SHOULD broadcast its rate limits to its peers using rate limits packet code (`0x14`). The rate limits is RLP-encoded information:
+
+```
+[ IP limits, PeerID limits, Topic limits ]
+```
+
+`IP limits`: 4-byte wide unsigned integer
+`PeerID limits`: 4-byte wide unsigned integer
+`Topic limits`: 4-byte wide unsigned integer
+
+The rate limits MAY also be sent as an optional parameter in the handshake.
+
+Each node SHOULD record rate limits advertised by its peers and adjust the number of sent packets in order not to exceed peer's rate limits. If the limit gets exceeded, the connection MAY be dropped.
```

Comment:
I'm not 100% sure what this means

Reply:
Assume Node A having Peer1, Peer2 and Pee3. Node A advertised rate limits of 5 messages per second per peer. Peer1 should throttle broadcasting messages to Node A and respect the limit of 5 messages per second. This brings a dangerous problem of skipping some messages.

Reply:
Can this be better formulated in the spec, not sure how yet.

Reply:
since whisper is queue based, maybe talking in those terms would be useful

Reply:
> Each node SHOULD respect rate limits advertised by its peers. The number of packets SHOULD be throttled in order not to exceed peer's rate limits. If the limit gets exceeded, the connection MAY be dropped by the peer.

Better?